sklearn/metrics/_plot/cap_curve.py

```diff
@@ -0,0 +1,521 @@
+            self.ax_.legend(loc="lower right")
+
+        return self
+
+    @classmethod
+    def from_predictions(
+        cls,
+        y_true,
+        y_pred,
+        *,
+        sample_weight=None,
+        pos_label=None,
+        normalize_scale=True,
+        plot_chance_level=True,
+        chance_level_kwargs=None,
+        plot_perfect=True,
+        perfect_level_kwargs=None,
+        despine=False,
+        name=None,
+        ax=None,
+        **kwargs,
+    ):
+        """Create the Cumulative Accuracy Profile.
+
+        This is also known as a (cumulative) gain curve. It can be used with
+        either classifiers or regressors (with positive outcomes only). Other
+        types of estimators will give rise to an error.
+
+        Parameters
+        ----------
+        y_true : array-like of shape (n_samples,)
+            True labels (for classifiers) or true values (for regressors).
+
+        y_pred : array-like of shape (n_samples,)
+            Target scores, can either be probability estimates of the positive
+            class, confidence values, non-thresholded measure of decisions
+            (as returned by "decision_function" on some classifiers) or the
+            "predict" output in the case of a regressor.
+
+        sample_weight : array-like of shape (n_samples,), default=None
+            Sample weights. If not `None`, the `y_true_cumulative` array is the
+            cumulative sum of the weights of the positive cases (ordered by
+            predictions) and `cumulative_total` is the cumulative sum of all
+            cases (ordered by predictions).
+
+        pos_label : int, float, bool or str, default=None
+            The label of the positive class. When `pos_label=None`, if `y_true`
+            is in {-1, 1} or {0, 1}, `pos_label` is set to 1, otherwise an
+            error will be raised. Should not be used in the case of regression.
+
+        normalize_scale : bool, default=True
+            Whether to normalize values between 0 and 1 for the plot.
+
+        plot_chance_level : bool, default=True
+            Whether to plot the expected curve of a classifier whose
+            predictions are independent of the feature values passed to
+            `decision_function` or `predict_proba`.
+
+        chance_level_kwargs : dict, default=None
+            Keyword arguments to be passed to matplotlib's `plot` for rendering
+            the chance level line.
+
+        plot_perfect : bool, default=True
+            Whether to plot the perfect model line.
+
+        perfect_level_kwargs : dict, default=None
+            Keyword arguments to be passed to matplotlib's `plot` for rendering
+            the perfect line.
+
+        despine : bool, default=False
+            Whether to remove the top and right spines from the plot.
+
+        name : str, default=None
+            Name of CAP curve for labeling. If `None`, name will be set to
+            `"Classifier"` or `"Regressor"`.
```

Comment:
I know we do this elsewhere, but I don't think these default names add much and I'd rather not have them. WDYT @glemaitre ?